In [403]:
import numpy as np
import pandas as pd

In [404]:
df = pd.read_csv('startup_funding.csv')
df.head()

Sr No Date dd/mm/yyyy                  Startup Name    Industry Vertical  \
0      1      09-01-2020                        BYJU’S               E-Tech   
1      2      13-01-2020                        Shuttl       Transportation   
2      3      09-01-2020                     Mamaearth           E-commerce   
3      4      02-01-2020  https://www.wealthbucket.in/              FinTech   
4      5      02-01-2020                        Fashor  Fashion and Apparel   

                             SubVertical City  Location  \
0                             E-learning      Bengaluru   
1              App based shuttle service        Gurgaon   
2  Retailer of baby and toddler products      Bengaluru   
3                      Online Investment      New Delhi   
4            Embroiled Clothes For Women         Mumbai   

              Investors Name       InvestmentnType Amount in USD Remarks  
0    Tiger Global Management  Private Equity Round  20,00,00,000     NaN  
1  Susquehanna Growth Equity              Series C     80,48,394     NaN  
2      Sequoia Capital India              Series B   1,83,58,860     NaN  
3             Vinod Khatumal          Pre-series A     30,00,000     NaN  
4    Sprout Venture Partners            Seed Round     18,00,000     NaN

In [405]:
df.shape

(3044, 10)

In [406]:
df['Remarks']

0                                                     NaN
1                                                     NaN
2                                                     NaN
3                                                     NaN
4                                                     NaN
5                                                     NaN
6                                                     NaN
7                                                     NaN
8                                                     NaN
9                                                     NaN
10                                                    NaN
11                                                    NaN
12                                                    NaN
13                                                    NaN
14                                                    NaN
15                                                    NaN
16                                                    NaN
17            

In [407]:
df.columns

Index(['Sr No', 'Date dd/mm/yyyy', 'Startup Name', 'Industry Vertical',
       'SubVertical', 'City  Location', 'Investors Name', 'InvestmentnType',
       'Amount in USD', 'Remarks'],
      dtype='object')

## 1. Summery of the dataset.
This dataset has funding information of the Indian startups from January 2015 to August 2017. It includes columns with the date funded, the city the startup is based out of, the names of the funders, and the amount invested (in USD).There are 3044 rows in the data and 10 columns.
<br> 
## 2. Columns/Meta data
1. ___Sr No___ - The serial numbering for the data.
2. ___date___ - The date on which the investment was received.
3. ___Startup Name___ - The name of the startup getting investment.
4. ___Industry Vertical___ - The industry of niche on which the startup is focusing.(eg. EdTech, Finance etc)
5. ___SubVertical___ - A more narrower area within a vertical, chosen to address very specific needs.
6. ___City Location___ - The city where the startup is based.
7. ___Investor Name___ - The name of the investor.
8. ___InvestmentnType___ - Tells Which round of the investment the current round is for the startup(private equity or seed funding).
9. ___Amount in USD___ - The amount of money received in the funding round in USD.
10. ___Remarks___ - It tells the type of private equity round, ie series A, series B etc. The startups that received seed funding has this column value as null/na.

## Problems with the dataset
1. Dirty data
- Many rows in the `Startup Name` column have special characters in their name. `validity`
- Startup `BYJU'S` have email or website in it. `validity` 
- Many rows in startup names column have .im, .com etc in their name. `validity`
- Entries from index 2108 to 2874 have shifted the Information of `Subvertical` to the `industry vertical` column. `validity`
- Due to shifting the information about the `Industry Vertical` of those startups is Not available. `completeness`
- Rows greater than index 2873, have No information about `Industry Vertical`, `Subvertical` and `City Location`. `completeness`
- Many Rows in the `Investor Name` column also have special characters in their names. `validity`
- Many Rows in the `Investment Type` column contain special characters. `validity`
- Many rows have not Disclosed the information about the `Investment Amount`. Have null values. `completeness`
- Rows that received Seed Funding Have Null in the `Remarks` column. `completeness`
- `Startup Name` have different names for same company. `consistancy`
- `Investor Name` have different name for same investor. `consistancy`
- `Industry Vertical` have different name for same niche. `consistancy`

name - 
`[5,69,241,469,472,474,617,688,689,695,714,715,721,740,743,744,
763,782,824,825,985,998,1062,1084,1155,1166,1212,1228,1260,1288,1358
,1396,1400,1434,1442,1469,1509,1510,1516,1526,1542,1555,1679,1719,1767
,1882,1936,2015,2020,2054,2077,2091,2189,2217,2224,2227,2236,2271,2319
,2335,2339,2349,2354,2363,2371,2376,2380,2407,2425,2443,2497,2500,2506
,2546,2562,2563,2604,2605,2606,2607,2608,2609,2610,2611,2612,2613,2622
,2655,2559,2705,2720,2722,2734,2740,2751,2782,2785,2796,2820,2846,2848
,2867,2875,2878,2880,2886,2919,2922,2928,2946,2948,2951,2956,2957,2969
,2971,2973,2978,2996,2997,3018,3026,3033,3035,3039,3040,3045]`

vertical - 
`[39,127,2022,2203,2222,2240,2366,2387,2396,2411,2421,2490,2510,2604,
2605,2606,2607,2608,2609,2610,2611,2612,2613,2649,2693,2826,2828,2841,]`
- shifting of  vertical and subvertical 2110-2874. Data not availabel from later 2874

Subvertical-
`[155,184,187,189,440,686,754,774,854,905,1217,1246,1320,1348,1415,1417,
1436,1464,1545,1555,1615,1688,1771,1783,1805,1848,1853,2059,2090,]`

city-
`[142,143,1352,1742.,1824.,1860,1913,1919,2071,2072,2164,2175,2225,2231
,2245,2257,2270,2321,2351,2451,2468,2504,2514,2595,2596,2604,2605,2606
,2607,2608,2609,2610,2611,2612,2613,2677,2683,2730,2814,2854,2869,]`




In [408]:
name = [5,69,241,469,472,474,617,688,689,695,714,715,721,740,743,744,763,782,824,825,985,998,1062,1084,1155,1166,1212,1228,1260,1288,1358,1396,1400,1434,1442,1469,1509,1510,1516,1526,1542,1555,1679,1719,1767,1882,1936,2015,2020,2054,2077,2091,2189,2217,2224,2227,2236,2271,2319,2335,2339,2349,2354,2363,2371,2376,2380,2407,2425,2443,2497,2500,2506,2546,2562,2563,2604,2605,2606,2607,2608,2609,2610,2611,2612,2613,2622,2655,2559,2705,2720,2722,2734,2740,2751,2782,2785,2796,2820,2846,2848,2867,2875,2878,2880,2886,2919,2922,2928,2946,2948,2951,2956,2957,2969,2971,2973,2978,2996,2997,3018,3026,3033,3035,3039,3040,3045]
len(name)

127

In [409]:
vertical = [39,127,2022,2203,2222,2240,2366,2387,2396,2411,2421,2490,2510,2604,2605,2606,2607,2608,2609,2610,2611,2612,2613,2649,2693,2826,2828,2841,]
# eCommmerce and ECommerce
# shifting of  vertical and subvertical 2110-2874. Data not availabel from later 2874

In [410]:
sub = [155,184,187,189,440,686,754,774,854,905,1217,1246,1320,1348,1415,1417,1436,1464,1545,1555,1615,1688,1771,1783,1805,1848,1853,2059,2090,]

In [411]:
city = [142,143,1352,1742.,1824.,1860,1913,1919,2071,2072,2164,2175,2225,2231,2245,2257,2270,2321,2351,2451,2468,2504,2514,2595,2596,2604,2605,2606,2607,2608,2609,2610,2611,2612,2613,2677,2683,2730,2814,2854,2869,]

In [412]:
df[df['Sr No']>2874]['Startup Name']

2874                          InstaLively
2875                                Vserv
2876                           Intruo.com
2877                              Niffler
2878                    CustomFurnish.com
2879                          MapMyGenome
2880    Crowdfire (formerly Justunfollow)
2881                            Bite Club
2882                           Localbanya
2883                               Appedo
2884                           Fitraq.com
2885                            ChargeBee
2886                        Groupon India
2887                            Process 9
2888                        MoonFrog Labs
2889                            Foodpanda
2890                                Paytm
2891                          DiabetOmics
2892                                  Qyk
2893                         Daily Rounds
2894                              Orobind
2895                              Routofy
2896                       Digital Gorkha
2897                             N

In [413]:
pd.set_option('display.max_columns',None)
pd.set_option('display.max_rows',None)


In [414]:
df[df['Sr No']<2110]['Industry Vertical'].unique()

array(['E-Tech', 'Transportation', 'E-commerce', 'FinTech',
       'Fashion and Apparel', 'Logistics', 'Hospitality', 'Technology',
       'E-Commerce', 'Aerospace', 'B2B-focused foodtech startup',
       'Finance', 'Video', 'Gaming', 'Software', 'Health and wellness',
       'Education', 'Food and Beverage', 'Health and Wellness',
       'B2B Marketing', 'Video Games', 'SaaS', 'Last Mile Transportation',
       'Healthcare', 'Customer Service', 'B2B', 'Consumer Goods',
       'Advertising, Marketing', 'IoT', 'Information Technology',
       'Consumer Technology', 'Accounting', 'Retail',
       'Customer Service Platform', 'Automotive', 'EdTech', 'Services',
       'Compliance', 'Transport', 'Artificial Intelligence', 'Tech',
       'Health Care', 'Luxury Label', 'Waste Management Service',
       'Deep-Tech', 'Agriculture', 'Energy', 'Digital Media', 'Saas',
       'Automobile', 'Agtech', 'Social Media', 'Fintech', 'Edtech', 'AI',
       'Ecommerce', 'Nanotechnology', 'Services Platfo

In [415]:
df.columns

Index(['Sr No', 'Date dd/mm/yyyy', 'Startup Name', 'Industry Vertical',
       'SubVertical', 'City  Location', 'Investors Name', 'InvestmentnType',
       'Amount in USD', 'Remarks'],
      dtype='object')

In [416]:
df[(df['Sr No']>=2109)&(df['Sr No']<=2873)]['Startup Name']

2108                                    CollegeDekho
2109                       Black White Orange Brands
2110                                        Biteclub
2111                                 Patel Logistics
2112                                           Wigzo
2113                                          Zapluk
2114                                 Godof Hospitals
2115                                     RoomsTonite
2116                                         Kaaryah
2117                                         Anakage
2118                                      EasyKhaana
2119                                          Shuttl
2120                                      Eckovation
2121                                       Fitcircle
2122                                          Applop
2123                                           CaRPM
2124                             NDTV Wedding portal
2125                                       Splitkart
2126                                         Z

In [417]:
temp = df.copy()

In [418]:
temp.columns

Index(['Sr No', 'Date dd/mm/yyyy', 'Startup Name', 'Industry Vertical',
       'SubVertical', 'City  Location', 'Investors Name', 'InvestmentnType',
       'Amount in USD', 'Remarks'],
      dtype='object')

In [419]:
## Renaming the columns
temp.rename(columns = ({
    'Sr No':'sno', 'Date dd/mm/yyyy':'date', 'Startup Name':'startup_name', 'Industry Vertical':'vertical',
       'SubVertical':'subvertical', 'City  Location':'city', 'Investors Name':'investor_name', 'InvestmentnType':'type',
       'Amount in USD':'amount', 'Remarks':'remarks'
}),inplace=True)

#### Lets first deal with the missing values.

In [420]:
# Industry vertical column
#shifting the vertical and subvertical column

temp.loc[(temp['sno']>2109) & (temp['sno']<2873),['vertical','subvertical']] = temp[(temp['sno']>2109) & (temp['sno']<2873)][['vertical','subvertical']].shift(1,axis=1)

In [421]:
temp[(temp['sno']>2109) & (temp['sno']<2873)]

sno                 date                                  startup_name  \
2109  2110           01-12-2015                     Black White Orange Brands   
2110  2111           01-12-2015                                      Biteclub   
2111  2112           01-12-2015                               Patel Logistics   
2112  2113           01-12-2015                                         Wigzo   
2113  2114           01-12-2015                                        Zapluk   
2114  2115           01-12-2015                               Godof Hospitals   
2115  2116           02-12-2015                                   RoomsTonite   
2116  2117           02-12-2015                                       Kaaryah   
2117  2118           03-12-2015                                       Anakage   
2118  2119           03-12-2015                                    EasyKhaana   
2119  2120           03-12-2015                                        Shuttl   
2120  2121           03-12-2015                                    Eckovation   
2121  2122           03-12-2015                                     Fitcircle   
2122  2123           04-12-2015                                        Applop   
2123  2124           04-12-2015                                         CaRPM   
2124  2125           04-12-2015                           NDTV Wedding portal   
2125  2126           04-12-2015                                     Splitkart   
2126  2127           07-12-2015                                       Zenatix   
2127  2128           07-12-2015                                 Customfurnish   
2128  2129           07-12-2015                                        Faasos   
2129  2130           07-12-2015                                         Vlurn   
2130  2131           07-12-2015                                Leaf Wearables   
2131  2132           07-12-2015                                     Coolshare   
2132  2133           08-12-2015                                      mFicient   
2133  2134           08-12-2015                                        HeyBiz   
2134  2135           08-12-2015                                        Yumist   
2135  2136           09-12-2015                                Jigsaw Academy   
2136  2137           09-12-2015                                      JoeHukum   
2137  2138           09-12-2015                                        Voxweb   
2138  2139           09-12-2015                                   Aye Finance   
2139  2140           09-12-2015                                   JumboBasket   
2140  2141           10-12-2015                                     BlackBuck   
2141  2142           10-12-2015                                    HomeStudio   
2142  2143           10-12-2015                                     PropheSee   
2143  2144           10-12-2015                                          i2e1   
2144  2145           10-12-2015                             Holisol Logistics   
2145  2146           10-12-2015                           Stay Simple Resorts   
2146  2147           10-12-2015                                       Paynear   
2147  2148           10-12-2015                                        Chillr   
2148  2149           11-12-2015                                        EureQa   
2149  2150           11-12-2015                                  LocalCircles   
2150  2151           11-12-2015                                      GetFocus   
2151  2152           11-12-2015                                     UrbanClap   
2152  2153           11-12-2015                                     OnlineRTI   
2153  2154           11-12-2015                                   Powerstores   
2154  2155           11-12-2015                                   Weddingz.in   
2155  2156           11-12-2015                        Intelligent Interfaces   
2156  2157           14-12-2015                                    Capricoast   
2157  2158           14-12-2015       

In [422]:
## Since the data about the vertical, subvertical and city is missing for many of the columns
## we are filling the data with some searching about the vertical and city. The subvertical empty values would
## be replaced with not available, as such sepecific data is difficult to find for such large number of companies.
(temp.isna().sum()/temp.shape[0])*100
## around 200 rows in the subvertical column

sno               0.000000
date              0.000000
startup_name      0.000000
vertical         30.683311
subvertical       5.683311
city              5.913272
investor_name     0.788436
type              0.131406
amount           31.537451
remarks          86.235217
dtype: float64

In [423]:
x = temp.copy()

##### After hitting my head and spending almost 3 hrs with AI tools to get the missing information reguarding Vertical of the startups.Even after giving to not assume the vertical value and answer completely on the basis of research, it kept on repeating and returned the value on the basis of frequency of the vertical and other previous information. Therefore, I get to know that the only way to get the credibal information about the Vertical is to search manually. But the data is missing for 30% or around 900 rows. Hence it would be impossible to research that information. Hence I am believeing these 🤬 tools with the data that they provided me.

In [424]:
industry_verticals_batch_1 = [
    "Advertising", "Food & Beverage", "Logistics", "SaaS / Marketing Tech", "Technology / SaaS", "Healthcare", "Travel Tech / Hospitality",
    "Technology / SaaS", "E-commerce", "Food Tech / Delivery", "Transportation", "EdTech", "Health and Wellness", "Technology / SaaS",
    "Technology (Automotive)", "Digital Media / Events", "E-commerce", "Energy", "Furniture E-commerce", "Food & Beverage", 
    "Technology / SaaS", "Health Tech / Wearables", "Social Media / Marketplaces", "Technology / SaaS", "SaaS / Business Services",
    "Food & Beverage", "EdTech", "Technology / SaaS", "Digital Media / Video", "FinTech"
]
#  index 2109 to 2138:

industry_verticals_batch_2 = [
    "E-commerce Grocery", "Transportation & Logistics", "Real Estate / Interior Design", "AI and Deep-Tech", "IT Services", "Logistics Tech", 
    "Hospitality", "FinTech / Payments", "FinTech / Wallets", "Technology / SaaS", "Social Media / Community Platform", "EdTech", 
    "Home Services / Marketplace", "Services / Government Transparency", "E-commerce / Retail", "Event Planning / Hospitality", 
    "B2B SaaS / Automation", "Real Estate", "Aviation / Travel Tech", "Transportation / Cab Aggregation", "SaaS / Business Intelligence", 
    "Event Management / SaaS", "Digital Media / News", "Video Streaming / SaaS", "Home Services", "Technology / SaaS", "Health Care / Telemedicine", 
    "Digital Media / Entertainment", "Logistics / Transportation", "EdTech / Online Learning",
]  # Industry verticals for startup indices 2139-2168

industry_verticals_batch_3 = [
    "Food Tech", "Automotive", "FinTech", "News Media", "Financial Tech", "Logistics", "E-commerce", "Digital Media", "Social Network",
    "Real Estate", "Travel Tech", "Real Estate", "Marketing Tech", "Transportation", "IT Services", "Social Media", "Retail Tech",
    "Video Streaming", "Healthcare", "Food & Beverage", "E-commerce", "Furniture E-commerce", "Online Marketplace", "Advertising",
    "Fitness", "Events Management", "Health Care", "EdTech", "Social Media", "Home Services"
]  # Industry verticals for startup indices 2169-2198
industry_verticals_batch_4 = [
    "E-commerce", "FinTech", "Social Media", "Logistics", "FinTech", "E-commerce", "Healthcare", "Marketing", "Technology",
    "SaaS", "Retail", "Fashion and Apparel", "FinTech", "EdTech", "Food & Beverage",
    "Real Estate", "SaaS", "Logistics", "Digital Media", "Fitness", "Hospitality", "Marketing", "EdTech", "E-commerce",
    "Technology", "Real Estate", "EdTech", "Healthcare", "Advertising", "Home Services",
]  # Industry verticals for startup indices 2199-2228
industry_verticals_batch_5 = [
    "Food & Beverage", "Insurance/FinTech", "Shared Mobility", "Sports Tech", "FinTech", "Education / Learning", "Video Games", "FinTech",
    "FinTech", "Education / Learning", "Cybersecurity", "SaaS", "Health Care", "Health Tech", "Home Services", "Furniture & Home Decor",
    "PayTech", "FinTech", "Retail Tech", "Travel Tech", "Real Estate Tech", "FinTech", "SaaS", "Healthcare", "Social Media", "Social Media",
    "Online Streaming", "Logistics & Delivery", "E-commerce", "Online Marketplace"
]  # Industry verticals for startup indices 2229-2258

industry_verticals_batch_6 = [
    "Home Services", "Furniture", "Technology / Analytics", "FinTech", "Tools & Equipment", "Education", "Health & Wellness",
    "E-commerce", "Healthcare", "EdTech", "Fitness", "Logistics", "FinTech", "Technology", "E-commerce",
    "Marketing", "Logistics", "Transportation", "Transportation", "Shared Mobility", "Technology", "Mobile Tech",
    "FinTech", "E-commerce", "Social Media", "Food & Beverage", "Health Tech", "Transportation", "FinTech", "Tax & Accounting"
]  # Industry verticals for startup indices 2259-2288
industry_verticals_batch_7 = [
    "FinTech", "B2B SaaS", "Marketplaces", "Fashion Retail", "Marketing", "Healthcare", "HR Tech", "Hospitality", "E-commerce", "Career Services",
    "Education", "B2B SaaS", "Health Care", "Social Networking", "Consumer Goods", "Food & Beverage", "Insurance", "Transportation", "FinTech",
    "Sports Tech", "Event Management", "Cashback & Coupons", "Recruitment", "Real Estate", "Health & Wellness", "Marketing", "Taxi & Transport",
    "Home Services", "Consumer Electronics", "Food & Beverage"
]  # Industry verticals for startup indices 2289-2318


industry_verticals_batch_8 = [
    "Health & Wellness", "FinTech", "Gaming", "Real Estate", "Travel Tech", "FinTech", "Social Media", "Media / Publishing", "Marketplace",
    "SaaS", "Online Marketplace", "Food & Beverage", "Legal / Compliance", "FinTech", "SaaS", "Consumer Goods", "E-commerce", "Health & Wellness",
    "Education", "Retail Tech", "Digital Media", "Fitness", "E-commerce", "EdTech", "Healthcare", "Consumer Internet", "Technology",
    "E-commerce", "Travel Tech", "Sports / Fitness"
]  # Industry verticals for startup indices 2319-2348
industry_verticals_batch_9 = [
    "Transportation", "E-commerce", "SaaS", "Sports Tech", "Hospitality", "Education / EdTech", "Consumer Services", "Technology", "FinTech",
    "SaaS", "Travel Tech", "Logistics", "Consulting / Services", "Food & Beverage", "Financial Services", "Event Management", "Healthcare",
    "Media / Entertainment", "Customer Service", "Retail", "FinTech", "Online Education", "E-commerce", "Consumer Electronics", "Technology",
    "SaaS", "E-commerce", "Healthcare", "Real Estate", "AdTech"
]  # Industry verticals for startup indices 2349-2378
industry_verticals_batch_10 = [
    "Social Networking", "Media", "Social Commerce", "Fashion and Apparel", "Real Estate", "E-commerce", "Social Media", "Digital Media",
    "Online Marketplace", "Healthcare", "Food & Beverage", "Telecommunications", "EdTech", "SaaS", "Customer Service",
    "Gaming", "Technology", "Advertising", "Online Marketplaces", "Healthcare", "Marketing", "SaaS", "Travel Tech",
    "AI & Machine Learning", "FinTech", "E-commerce", "Travel Tech", "Health & Wellness", "Real Estate", "Retail Tech"
]  # Industry verticals for startup indices 2379-2408
industry_verticals_batch_11 = [
    "E-commerce", "EdTech", "Social Media", "Technology", "SaaS", "Marketing", "FinTech", "Healthcare", "Real Estate", "Logistics",
    "Retail", "Food & Beverage", "Technology", "SaaS", "Home Services", "Gaming", "Finance", "Transportation", "Health & Wellness", "E-commerce",
    "Digital Media", "EdTech", "Travel Tech", "AI / Deep Tech", "Home Decor", "B2B SaaS", "Media", "Social Media", "FinTech", "Consumer Goods"
]  # Industry verticals for startup indices 2409-2438
industry_verticals_batch_12 = [
    "E-commerce", "FinTech", "EdTech", "Food Tech", "Energy", "Health Care", "Real Estate", "Digital Media", "Fitness",
    "E-commerce", "Marketing", "Travel Tech", "Automotive", "Education", "SaaS", "Consumer Goods", "B2B SaaS", "Healthcare",
    "E-commerce", "Education", "Technology", "FinTech", "SaaS", "Real Estate", "Digital Media", "Healthcare", "Marketing",
    "Logistics", "Retail", "Technology"
]  # Industry verticals for startup indices 2439-2468
industry_verticals_batch_13 = [
    "Education", "E-commerce", "Retail", "Healthcare", "Logistics", "Food Tech", "Customer Service", "E-commerce", "SaaS",
    "Health & Wellness", "Social Media", "E-commerce", "Healthcare", "Retail Tech", "Digital Media", "EdTech", "SaaS",
    "Real Estate", "FinTech", "Health Care", "Social Media", "Marketing", "SaaS", "Logistics", "Education", "Transportation",
    "Technology", "Gaming", "Logistics", "E-commerce"
]  # Industry verticals for startup indices 2469-2498

industry_verticals_batch_14 = [
    "E-commerce", "Transportation", "E-commerce", "Data Analytics", "SaaS", "E-commerce", "Printing & Design", "FinTech / Payments",
    "Logistics & Supply Chain", "Interior Design", "Food & Beverage", "AgroTech", "SaaS / Digital Media", "Software Services", "Consumer Services",
    "E-commerce / Grocery", "Social Media", "FinTech", "Food Delivery", "Financial Services", "Media / Publishing", "Logistics Tech", "Beauty & Fashion",
    "EdTech", "Digital Media", "AgroTech", "Healthcare", "Logistics", "Consumer Electronics", "Video Social Media", "E-commerce", "AI & Analytics",
    "FinTech", "SaaS", "Fashion & Accessories", "Food Services", "Home Services", "Furniture E-commerce", "EdTech", "Real Estate", "E-commerce",
    "Food & Beverages", "Food Services", "Event Management", "Corporate Services", "Transportation", "Mobile Apps", "Food Services", "Digital Advertising", "Fashion & Lifestyle"
]  # Industry verticals for startup indices 2499-2548
industry_verticals_batch_15 = [
    "Logistics", "FoodTech", "Safety & Security", "Fitness", "Restaurant Tech", "Hospitality", "Food & Beverage", "E-commerce", "Education",
    "Consumer Goods", "Food & Beverage", "Digital Security", "Coaching & Learning", "Hospitality", "FinTech", "Payments", "Food Delivery",
    "EdTech", "Telecom", "Transportation", "Digital Media", "Media", "FoodTech", "Logistics", "E-commerce", "HR Tech", "Real Estate",
    "FinTech", "Mobile Apps", "E-commerce", "Gaming", "Food Delivery", "Analytics", "Logistics", "HR Tech", "Online Learning", "Retail Tech",
    "Health & Wellness", "FoodTech", "Furniture E-commerce", "SaaS", "E-commerce", "Health & Wellness", "Logistics", "Real Estate", "Travel Tech",
    "Hospitality", "E-commerce", "Social Media", "E-commerce"
]  # Industry verticals for startup indices 2549-2598
industry_verticals_batch_16 = [
    "FinTech", "Healthcare", "Car Rentals", "Media", "Jewelry", "E-commerce", "Education", "Transportation", "Hospitality", "Online Marketplace",
    "E-commerce", "Media & Entertainment", "FinTech", "Logistics", "Rental & Sharing Economy", "Food Delivery", "E-commerce", "Online Marketplace",
    "Retail Tech", "Logistics", "Online Education", "Food Delivery", "Media Streaming", "Social Media", "EdTech", "Travel Tech", "Marketplace",
    "Real Estate", "E-commerce", "Gaming", "Consumer Services", "Furniture E-commerce", "Retail", "FinTech", "Real Estate", "E-commerce",
    "Food & Beverage", "Home Services", "Education", "Food Delivery", "HR Tech", "Health & Wellness", "E-commerce", "Digital Media", "Logistics",
    "E-commerce", "Hospitality", "Logistics", "Fashion & Apparel", "E-commerce"
]  # Industry verticals for startup indices 2599-2648
industry_verticals_batch_17 = [
    "E-commerce", "Retail Tech", "EdTech", "Health Tech", "Technology Consulting", "E-commerce", "Food Delivery", "FinTech", "Print & Stationery",
    "Sustainable Tech", "Event Management", "Visual Content", "Travel Tech", "E-commerce", "Automotive", "Waste Management", "FinTech", "Parenting",
    "E-commerce", "Price Comparison", "E-commerce", "Marketplace", "Food & Beverage", "Gaming", "Social Networking", "Travel Tech", "E-commerce",
    "Renewable Energy", "Healthcare", "Shared Mobility", "Social Commerce", "E-commerce", "Kitchen & Food Appliances", "E-commerce", "Healthcare",
    "Online Gaming", "E-commerce", "Marketplaces", "B2B SaaS", "Health & Wellness", "Technology", "E-commerce", "Education", "Healthcare", "Real Estate",
    "Furniture & Decor", "Logistics", "EdTech", "HR Tech", "Online Marketplace"
]  # Industry verticals for startup indices 2649-2698
industry_verticals_batch_18 = [
    "E-commerce", "Travel Tech", "Fashion & Apparel", "Food Delivery", "Food & Beverage", "Education", "Event Management", "FinTech", "Customer Support",
    "E-commerce", "Logistics", "E-commerce", "SaaS", "Home Services", "Health Tech", "Marketplace", "Delivery & Transportation", "E-commerce",
    "Healthcare", "Social Networking", "Transportation", "B2B SaaS", "Education", "Healthcare", "MarTech", "Retail Tech", "E-commerce",
    "Marketplace", "Health & Wellness", "Logistics", "FinTech", "Retail Tech", "Education", "E-commerce", "Health Tech", "FinTech",
    "Digital Media", "E-commerce", "Health & Wellness", "Marketplace", "E-commerce", "E-commerce", "E-commerce", "Hotel & Travel", "Food Delivery",
    "E-commerce", "Mobility", "Healthcare", "Food & Beverage", "Electric Vehicles"
]  # Industry verticals for startup indices 2699-2748
industry_verticals_batch_19 = [
    "Home Services", "Social Media", "Health Tech", "Transportation", "Gaming", "E-commerce", "Payment Solutions", "FinTech", "Events",
    "E-commerce", "Entertainment", "Digital Media", "Transportation", "Subscription Services", "Logistics", "Food Delivery", "Rental Services",
    "FinTech", "Digital Marketing", "Online Marketplace", "Health & Wellness", "Travel Tech", "E-commerce", "Education", "E-commerce",
    "Media", "E-commerce", "Social Media", "Logistics", "Digital Advertising", "Retail", "Online Marketplace", "E-commerce", "Travel Tech",
    "Fitness", "Healthcare", "E-commerce", "Travel Tech", "Marketplace", "SaaS", "Education", "Health Tech", "Social Media", "Real Estate",
    "E-commerce", "E-commerce", "E-commerce", "Social Commerce", "Online Marketplace", "Health & Wellness"
]  # Industry verticals for startup indices 2749-2798

industry_verticals_batch_20 = [
    "Payments", "Education", "SaaS", "Real Estate", "Travel Tech", "E-commerce", "Digital Media", "Transportation", "Logistics",
    "E-commerce", "Gaming & Entertainment", "Social Media", "E-commerce", "Logistics", "E-commerce", "E-commerce", "HR Tech",
    "Health Tech", "Transportation", "Real Estate", "Fashion & Lifestyle", "E-commerce", "Healthcare", "Educational Services",
    "Food Delivery", "Food & Beverage", "FinTech", "SaaS", "E-commerce", "MarTech", "Consumer Services", "Consumer Goods",
    "E-commerce", "SaaS", "Health Tech", "Financial Services", "E-commerce", "Hospitality", "Social Media", "Real Estate",
    "Finance & Lending", "Online Marketplace", "Health & Wellness", "E-commerce", "Logistics", "FinTech", "E-commerce", "HR Tech", "SaaS", "Education"
]  # Industry verticals for startup indices 2799-2848

industry_verticals_batch_21 = [
    "EdTech", "Travel Tech", "Food & Beverage", "E-commerce", "E-commerce", "Education", "HR Tech", "FinTech", "Legal Tech",
    "Real Estate", "Marketing", "Health & Wellness", "E-commerce", "Fitness", "Consumer Goods", "SaaS", "FinTech",
    "E-commerce", "Marketplace", "Logistics", "Social Media", "E-commerce", "Rental Services", "Education", "Marketing"
]  # Industry verticals for startup indices 2849-2873
industry_verticals_batch_22 = [
    "Advertising", "Advertising", "Technology", "Technology", "E-commerce", "Healthcare", "Software", "Food & Beverage", "E-commerce", "Technology",
    "SaaS", "E-commerce", "Customer Service Platform", "Video Games", "Food & Beverage", "FinTech", "Healthcare", "Technology", "Healthcare", "Healthcare",
    "Travel Tech", "Marketing", "Real Estate", "Digital Media", "Digital Media", "Transportation", "Sports", "Information Technology", "Transportation", "Travel Tech",
    "Digital Media", "Healthcare Technology", "Services Platform", "Technology", "SaaS", "Automotive", "Gaming", "FinTech", "Furniture E-commerce", "Food & Beverage",
    "Technology", "E-commerce", "Technology", "E-commerce", "Education", "Hospitality", "Social Media", "Video", "Healthcare", "Energy", "FinTech", "Digital Media",
    "Fashion and Apparel", "Aerospace", "Finance", "FinTech", "Travel Tech", "Technology", "Software", "Technology", "Technology", "Food & Beverage",
    "Online Marketplace", "Marketing", "Technology", "Software", "Food & Beverage", "Technology", "E-commerce", "Food & Beverage", "E-commerce", "Analytics",
    "E-commerce", "Digital Media", "FinTech", "Fashion and Apparel", "FinTech", "Digital Media", "Services Platform", "Automotive", "Food Tech", "E-commerce",
    "Technology", "FinTech", "IoT", "SaaS", "E-commerce", "Travel Tech", "Healthcare", "SaaS", "Fashion and Apparel", "FinTech", "Digital Media", "Travel Tech", "Technology",
    "E-commerce", "Education", "Technology", "Software", "SaaS", "Real Estate", "Travel Tech", "Travel Tech", "Real Estate", "Consumer Goods", "Online Education", "B2B",
    "FinTech", "Food Tech", "Food & Beverages", "B2B", "B2B", "Food Tech", "Food Tech", "E-commerce", "Healthcare", "Fashion and Apparel", "Technology", "Social Media",
    "Fitness", "E-commerce", "E-commerce", "FinTech", "SaaS", "Social Media", "FinTech", "EdTech", "Fashion and Apparel", "Healthcare", "Real Estate", "Marketing",
    "E-commerce", "Food & Beverage", "Healthcare", "E-commerce", "Marketing", "Travel Tech", "Crowdfunding / Finance", "Healthcare", "E-commerce", "Marketing",
    "Marketing", "Health and Wellness", "Hospitality", "Sports / Fitness", "Education", "SaaS", "Finance", "Entertainment / Toys", "Digital Media", "Health and Wellness",
    "Online Education", "E-commerce", "EdTech", "Social Impact / Agtech", "B2B", "Food & Beverage", "Education", "AI / Deep Tech", "Fashion and Apparel", "Employment / Job Platform",
    "Education", "Gaming / Entertainment", "Automotive", "Real Estate", "Printing & Personalized Gifting", "Clean-tech / Nanotechnology", "Artificial Intelligence", "Hospitality / SaaS", 
    "Marketing / Coupon Deals"
]
# 2874 to 3043:

In [425]:
industry_verticals_all = []
for i in range(1, 23):
    industry_verticals_all += globals()[f"industry_verticals_batch_{i}"]


In [426]:
len(industry_verticals_all)

935

In [427]:
x.loc[x.index>=2109,'vertical'] = industry_verticals_all
(x.isna().sum()/temp.shape[0])*100

sno               0.000000
date              0.000000
startup_name      0.000000
vertical          0.000000
subvertical       5.683311
city              5.913272
investor_name     0.788436
type              0.131406
amount           31.537451
remarks          86.235217
dtype: float64

#### Replacing 5% missing values in subvertical and city column with Not Available.

In [428]:
temp2 = x.copy()

In [429]:
temp2['subvertical'] = temp2['subvertical'].fillna('Not Available')

In [430]:
temp2['city'] = temp2['city'].fillna('Not Available')

#### Replacing the Missing values in the investor_name and amount with Not Disclosed.

In [431]:
temp2['investor_name'] = temp2['investor_name'].fillna('Not Disclosed')

In [432]:
# encoding the not disclosed amount with -1 as the datatype will be numerical
temp2['amount'] = temp2['amount'].fillna('-1')

In [433]:
temp2.isna().sum()

sno                 0
date                0
startup_name        0
vertical            0
subvertical         0
city                0
investor_name       0
type                4
amount              0
remarks          2625
dtype: int64

In [434]:
# The remarks column contains the information about which series round is the private funding, also more than 75% of the data is missing.
# Hence dropping the column would be the best action.

In [435]:
temp2 = temp2.drop(columns=('remarks'))

In [436]:
#### 
temp2['startup_name'] = temp['startup_name']

In [437]:
## startup_name column
temp2['startup_name'].value_counts()

startup_name
Ola Cabs                                                  8
Swiggy                                                    8
Paytm                                                     7
Meesho                                                    6
Nykaa                                                     6
NoBroker                                                  6
Medinfi                                                   6
UrbanClap                                                 6
Flipkart                                                  5
Grofers                                                   5
Uniphore                                                  5
Capital Float                                             5
Moglix                                                    5
Toppr                                                     5
Jugnoo                                                    5
Icertis                                                   4
Wooplr                     

In [438]:
# converting all names to lower case
temp2['startup_name'] = temp2['startup_name'].str.strip().str.lower()

In [439]:
# extracting names from links

In [440]:
temp2[temp2['startup_name'].str.contains('www.')]

sno        date                  startup_name vertical        subvertical  \
3    4  02-01-2020  https://www.wealthbucket.in/  FinTech  Online Investment   

        city   investor_name          type     amount  
3  New Delhi  Vinod Khatumal  Pre-series A  30,00,000

In [441]:
temp2.loc[3,'startup_name'] = 'wealthbucket'

In [442]:
temp2.loc[3,'startup_name']

'wealthbucket'

In [443]:
# removing .com, .in, .io from names
temp2['startup_name'] = temp2['startup_name'].str.replace('.com','').str.replace('.in','').str.replace('.io','')

In [444]:
# correcting the incnsistancy with byju's name
temp2.loc[temp2['startup_name'].str.contains('byju'),'startup_name'] = 'byju’s'

In [445]:
temp2['startup_name'].value_counts()

startup_name
ola cabs                                                  8
swiggy                                                    8
byju’s                                                    7
paytm                                                     7
lenskart                                                  7
lendingkart                                               6
nykaa                                                     6
healthifyme                                               6
holachef                                                  6
medinfi                                                   6
flipkart                                                  6
urbanclap                                                 6
toppr                                                     6
cardekho                                                  6
blackbuck                                                 6
meesho                                                    6
nobroker                   

#### Other cleaning

In [446]:
temp2.loc[temp2['startup_name'].str.contains('zebi'),'startup_name'] = temp2[temp2['startup_name'].str.contains('zebi')]['startup_name'].str.replace(' (yet to launch)','')

In [447]:
# \\xe2\\x80\\x99 -> "'"
# \\n -> ""
# \\xc2\\xa0 -> ""
# \  -> ""
temp2['startup_name'] = temp2['startup_name'].str.replace(r'\\xe2\\x80\\x99',"'").str.replace(r'\\n',' ').str.replace(r'\\xc2\\xa0','')

In [448]:
temp2.loc[temp2['startup_name'].str.contains(r'\('),['startup_name']] = ['nearbuy','1mg','crowdfire']

#### Vertical Column

In [449]:
temp['vertical'].unique()

array(['E-Tech', 'Transportation', 'E-commerce', 'FinTech',
       'Fashion and Apparel', 'Logistics', 'Hospitality', 'Technology',
       'E-Commerce', 'Aerospace', 'B2B-focused foodtech startup',
       'Finance', 'Video', 'Gaming', 'Software', 'Health and wellness',
       'Education', 'Food and Beverage', 'Health and Wellness',
       'B2B Marketing', 'Video Games', 'SaaS', 'Last Mile Transportation',
       'Healthcare', 'Customer Service', 'B2B', 'Consumer Goods',
       'Advertising, Marketing', 'IoT', 'Information Technology',
       'Consumer Technology', 'Accounting', 'Retail',
       'Customer Service Platform', 'Automotive', 'EdTech', 'Services',
       'Compliance', 'Transport', 'Artificial Intelligence', 'Tech',
       'Health Care', 'Luxury Label', 'Waste Management Service',
       'Deep-Tech', 'Agriculture', 'Energy', 'Digital Media', 'Saas',
       'Automobile', 'Agtech', 'Social Media', 'Fintech', 'Edtech', 'AI',
       'Ecommerce', 'Nanotechnology', 'Services Platfo

In [450]:
replace_dict = {
    # E-Commerce variations
    'Ecommerce': 'E-Commerce',
    'E-commerce': 'E-Commerce',
    'eCommerce': 'E-Commerce',
    'eCommece': 'E-Commerce',
    'E-Commerce': 'E-Commerce',
    'ECommerce': 'E-Commerce',
    'ecommerce': 'E-Commerce',

    # FinTech variations
    'Fintech': 'FinTech',
    'Financial Tech': 'FinTech',
    'Fin-Tech': 'FinTech',
    'FinTech': 'FinTech',
    'Insurance/FinTech': 'FinTech',
    'FinTech / Wallets': 'FinTech',

    # EdTech variations
    'Online Education Information platform': 'Online Education',
    ''
    'Edtech': 'EdTech',
    'Ed-tech': 'EdTech',
    'EdTech': 'EdTech',
    'Online Education': 'EdTech',
    'Education / EdTech': 'EdTech',
    'EdTech / Online Learning': 'EdTech',

    # SaaS variations
    'Saas': 'SaaS',
    'SaaS': 'SaaS',
    'Technology / SaaS': 'SaaS',
    'SaaS / Business Services': 'SaaS',
    'SaaS / Marketing Tech': 'SaaS',

    # Consumer Internet
    'Consumer internet': 'Consumer Internet',
    'Consumer Interne': 'Consumer Internet',
    'Consumer Internet': 'Consumer Internet',

    # Food & Beverages
    'Food & Beverages': 'Food & Beverages',
    'Food & Beverage': 'Food & Beverages',
    'Food and Beverages': 'Food & Beverages',
    'Food-Tech': 'Food Tech',
    'B2B-focused foodtech startup': 'Food Tech',
    'Food Tech': 'Food Tech',
    'FoodTech': 'Food Tech',
    'Food Tech / Delivery': 'Food Tech',
    'Food Services': 'Food Tech',
    'Food': 'Food',

    # Healthcare
    'Health and wellness': 'Healthcare',
    'Health and Wellness': 'Healthcare',
    'Health and Wellness': 'Healthcare',
    'healthcare': 'Healthcare',
    'Health Care': 'Healthcare',
    'Healthcare': 'Healthcare',
    'Health Tech / Wearables': 'Health Tech',
    'Healthcare Technology': 'Health Tech',

    # Technology
    'E-Tech': 'Technology',
    'Deep-Tech': 'Deep-Tech',
    'Clean-tech': 'Technology',
    'Tech': 'Technology',
    'Technology': 'Technology',
    'IT': 'Technology',
    'Technology (Automotive)': 'Technology',
    'Information Technology': 'Technology',
    'IT Services': 'Technology',

    # Miscellaneous corrections / typos
    'Fiinance': 'Finance',
    'eCommece': 'E-Commerce',
    'Video': 'Video',
    'Video Games': 'Video Games',
    'Customer Service': 'Customer Service',
    'Customer Service Platform': 'Customer Service',
    'Services': 'Services',
    'Services Platform': 'Services',
    'Logistics Tech': 'Logistics',
    'Transportation': 'Logistics',
    'Last Mile Transportation': 'Logistics',
    'Transport': 'Logistics',
    'Logistics & Delivery': 'Logistics',
    'Taxi & Transport': 'Logistics',
    'Travel Tech': 'Logistics',
    'Transportation & Logistics': 'Logistics',
    'Online Education Information platform': 'Online Education',
    'Inspiration': 'Media / Content',
    'Storytelling': 'Media / Content',
    'Lifestyle': 'Media / Content',
    'Media': 'Media / Content',

    # Domains / generic categories
    'Others': 'Others',
    None: 'Unknown',
    'nan': 'Unknown',
    'Education / Learning' : 'Education',
    'Media / Entertainment' : 'Media & Entertainment',
    'AI / Deep Tech':'Artificial Intelligence',
    'AI and Deep-Tech':'Artificial Intelligence',
    'AI':'Artificial Intelligence',
    'AI & Machine Learning':'Artificial Intelligence',
    'B2B Platform':'B2B',
    'B2B Marketing':'B2B',
    'Real Estate Tech':'',
    'PayTech':'Payment',
    'Payment Solutions':'Payment',
    'Payment':'Payment',
    'Tax & Accounting':'Accounting',
    'Travel Tech / Hospitality':'Hospitality',
    'Digital Media / Entertainment':'Entertainment',
    'Entertainment / Toys':'Toys',
    'Media & Entertainment':'Entertainment',
    'Gaming / Entertainment':'Entertainment',
    'Video Games':'Gaming',
    'Digital Media / Events':'Digital Media',
    'Digital Media / Video':'Digital Media',
    'Social Media / Marketplaces':'Social Media',
    'Social Media / Community Platform':'Social Media',
    'Real Estate / Interior Design': 'Real Estate',
    'B2B SaaS / Automation':'B2B SaaS',
    'Online Gaming':'Gaming',
    
    
}


In [451]:
# Now most of the verticals have been categorised.

temp2['vertical'] = temp2['vertical'].replace(replace_dict)

In [452]:
# For the subvertical column, it is very specific and also not of much use.

#### Lets first deal with the amount column.

In [453]:
# clean the amount column and change the amount from dollors to Cr.

In [454]:
temp2.loc[temp2['amount']=='undisclosed','amount'] = '-1'

In [455]:
temp2.loc[temp2['amount']=='unknown','amount'] = '-1'

In [456]:
temp2.loc[temp2['amount']=='Undisclosed','amount'] = '-1'

In [462]:
temp2.loc[temp2['amount']=='N/A','amount'] = '-1'

In [463]:
temp2.loc[temp2['amount']=='14,342,000+','amount'] = '-1'

In [464]:
# \\xc2\\xa0
temp2['amount'] = temp2['amount'].str.replace(r'\\xc2\\xa0','')

In [465]:

temp2['amount'] = (temp2['amount'].str.replace(',','').astype(float)*(88.67))/10000000

In [466]:
temp2.loc[temp2['amount']<0,'amount'] = 0

#### Type column

In [467]:
temp2['type'].value_counts()

type
Private Equity                 1356
Seed Funding                   1355
Seed/ Angel Funding              60
Seed / Angel Funding             47
Seed\\nFunding                   30
Debt Funding                     25
Series A                         24
Seed/Angel Funding               23
Series B                         20
Series C                         14
Series D                         12
Angel / Seed Funding              8
Seed Round                        7
Pre-Series A                      4
Private Equity Round              4
Seed                              4
Seed / Angle Funding              3
pre-Series A                      2
Venture Round                     2
Equity                            2
Corporate Round                   2
Series E                          2
Series F                          2
Private                           1
Debt-Funding                      1
Term Loan                         1
Seed funding                      1
PrivateEquity          

In [468]:
temp2.loc[temp2['type'].str.contains(r'^Series',na=False),'type'] = 'Private Equity'

In [469]:
temp2['type'] = temp2['type'].str.replace(r'\\n',' ')

In [470]:
replace_dict = {
    # Seed & Angel
    "Seed Funding": "Seed", "Seed\\nFunding": "Seed", "Seed funding": "Seed",
    "Seed": "Seed", "Seed Round": "Seed", "Seed Funding Round": "Seed",
    "Seed/ Angel Funding": "Seed", "Seed / Angel Funding": "Seed",
    "Seed/Angel Funding": "Seed", "Angel / Seed Funding": "Seed",
    "Seed / Angle Funding": "Seed", "Angel Funding": "Seed",
    "Angel Round": "Seed", "Angel": "Seed",
    
    # Pre-Series A
    "Pre-Series A": "Pre-Series A", "Pre Series A": "Pre-Series A",
    "pre-Series A": "Pre-Series A", "pre-series A": "Pre-Series A",
    "Pre-series A": "Pre-Series A",
    
    # Private Equity
    "Private Equity": "Private Equity", "PrivateEquity": "Private Equity",
    "Private\\nEquity": "Private Equity", "Private Equity Round": "Private Equity",
    "Private Funding": "Private Equity", "Private": "Private Equity",
    
    # Debt
    "Debt Funding": "Debt", "Debt-Funding": "Debt", "Debt": "Debt",
    "Structured Debt": "Debt", "Term Loan": "Debt",
    "Debt and Preference capital": "Debt",
    
    # Venture
    "Venture Round": "Venture Round", "Venture - Series Unknown": "Venture Round",
    "Venture": "Venture Round", "Single Venture": "Venture Round",
    
    # Corporate
    "Corporate Round": "Corporate Round",
    
    # Crowd
    "Crowd funding": "Crowdfunding", "Crowd Funding": "Crowdfunding",
    
    # Equity & Misc
    "Equity": "Equity (Other)", "Equity Based Funding": "Equity (Other)",
    "Maiden Round": "Other", "Funding Round": "Other"
}

temp2["type"] = temp2["type"].replace(replace_dict)


In [471]:
temp2['type']

0        Private Equity
1        Private Equity
2        Private Equity
3          Pre-Series A
4                  Seed
5        Private Equity
6        Private Equity
7        Private Equity
8        Private Equity
9                  Seed
10       Private Equity
11       Private Equity
12       Private Equity
13                 Debt
14                 Seed
15                 Seed
16       Private Equity
17       Private Equity
18       Private Equity
19                 Seed
20        Venture Round
21       Private Equity
22                 Debt
23                 Debt
24                 Seed
25       Private Equity
26                 Seed
27                 Seed
28       Private Equity
29       Private Equity
30                  NaN
31                Other
32       Private Equity
33       Private Equity
34      Corporate Round
35                Other
36       Private Equity
37       Private Equity
38         Pre-Series A
39       Private Equity
40                 Seed
41       Private

In [472]:
temp2.head()

sno        date  startup_name             vertical  \
0    1  09-01-2020        byju’s           Technology   
1    2  13-01-2020        shuttl            Logistics   
2    3  09-01-2020     mamaearth           E-Commerce   
3    4  02-01-2020  wealthbucket              FinTech   
4    5  02-01-2020        fashor  Fashion and Apparel   

                             subvertical       city  \
0                             E-learning  Bengaluru   
1              App based shuttle service    Gurgaon   
2  Retailer of baby and toddler products  Bengaluru   
3                      Online Investment  New Delhi   
4            Embroiled Clothes For Women     Mumbai   

               investor_name            type       amount  
0    Tiger Global Management  Private Equity  1773.400000  
1  Susquehanna Growth Equity  Private Equity    71.365110  
2      Sequoia Capital India  Private Equity   162.788012  
3             Vinod Khatumal    Pre-Series A    26.601000  
4    Sprout Venture Partners            Seed    15.960600

#### city column

In [473]:
temp2['city'] = temp2['city'].str.replace(r'\\xc2\\xa0','')

In [474]:
temp2['city'] = temp2['city'].str.split('/').str.get(0)

In [475]:
temp2['city'] = temp2['city'].str.split(',').str.get(0)

In [476]:
temp[temp2['city'].isin(['SFO','seattle','New York'])]

sno        date startup_name   vertical              subvertical  \
86      87  06-06-2019   Biz2Credit    FinTech  Online Lending Platform   
131    132  12-11-2018    Blackbuck  Food Tech  POS For Online Ordering   
2349  2350  21-10-2015   AgentDesks       None   Real Estate Mobile CRM   

                     city                                   investor_name  \
86               New York                              WestBridge Capital   
131   New York, Bengaluru                                    Tiger Global   
2349      New York/ India  Sierra Ventures, Cota Capital, Vegas Tech Fund   

              type       amount remarks  
86        Series B  5,20,00,000     NaN  
131       Series A    10,00,000     NaN  
2349  Seed Funding    29,50,000     NaN

In [477]:
temp2.loc[temp2.index==131,'city'] = 'Bengaluru'

In [478]:
temp2['city'] = temp2['city'].str.strip()

In [479]:
temp2['city'] = temp2['city'].replace('Nw Delhi','New Delhi')

In [480]:
temp2['city'] = temp2['city'].replace('Delhi & Cambridge','New Delhi')

In [481]:
temp2['city'] = temp2['city'].replace('Bengaluru and Gurugram','Bengaluru')

In [482]:
temp2['city'] = temp2['city'].replace('Bengaluru and Gurugram','Bengaluru')

In [483]:
temp2['city'] = temp2['city'].replace('SFO','Bengaluru')

In [484]:
temp2['city'].value_counts()

city
Bangalore        708
Mumbai           573
New Delhi        430
Gurgaon          292
Not Available    180
Bengaluru        144
Pune             112
Hyderabad        100
Chennai           98
Noida             94
Gurugram          50
Ahmedabad         38
Delhi             34
Jaipur            30
Kolkata           21
Indore            13
Goa               11
Chandigarh        11
Vadodara          10
Singapore          8
Coimbatore         5
India              5
Kanpur             4
Bhopal             3
Faridabad          3
Nagpur             3
Kormangala         2
Udaipur            2
Agra               2
Trivandrum         2
Kochi              2
Udupi              2
Gwalior            2
Surat              2
US                 2
San Francisco      2
Ahemadabad         2
USA                2
New York           2
Siliguri           2
Hubli              1
Lucknow            1
London             1
Kerala             1
Kozhikode          1
Varanasi           1
Karur              1
Jodhpur 

#### investor_name column

In [485]:
temp2['investor_name'].value_counts()

investor_name
Undisclosed Investors                                                                                                                                                                39
Undisclosed investors                                                                                                                                                                30
Ratan Tata                                                                                                                                                                           25
Not Disclosed                                                                                                                                                                        25
Indian Angel Network                                                                                                                                                                 23
Kalaari Capital                                                   

In [486]:
# we will take only the first investor form the list of investors and remove abbreviations like & others etc 

In [487]:
other = {
    'and others':'',
    '& others':'',
    'and Others':'',
    '& Others':'',
    'and other':'',
    '& other':'',
    'and Other':'',
    '& Other':'',
}

In [488]:
temp2['investor_name'] = temp2['investor_name'].str.split(',').str.get(0).replace(other)

In [489]:
temp2['investor_name'] = temp2['investor_name'].str.replace(r'\\xe2\\x80\\x99','')


In [490]:
# remove inconsistancy in the names of prominent investors.

In [491]:
temp2.loc[temp2['investor_name'].str.contains('Accel '),'investor_name'] = 'Accel Partners'

In [492]:
temp2.loc[temp2['investor_name'].str.contains('equoia'),'investor_name'] = 'Sequoia Capital'

In [493]:
temp2.loc[temp2['investor_name'].str.contains('Kalaari '),'investor_name'] = 'Kalaari Capital'

In [494]:
temp2.loc[temp2['investor_name'].str.contains('ndisclosed'),'investor_name'] = 'Not Disclosed'

In [495]:
temp2.loc[temp2['investor_name'].str.contains('Tata'),'investor_name'] = 'Ratan Tata'

In [496]:
temp2.loc[temp2['investor_name'].str.contains('Indian Angel'),'investor_name'] = 'Indian Angel Network'

In [497]:
temp2.loc[temp2['investor_name'].str.contains('Blume'),'investor_name'] = 'Blume Ventures'

In [498]:
temp2.loc[temp2['investor_name'].str.contains('SAIF'),'investor_name'] = 'SAIF Partners'

In [499]:
temp2.loc[temp2['investor_name'].str.contains('Tiger'),'investor_name'] = 'Tiger Global'

In [500]:
temp2.loc[temp2['investor_name'].str.contains('IDG'),'investor_name'] = 'IDG Ventures'

In [501]:
temp2.loc[temp2['investor_name'].str.contains('Matrix'),'investor_name'] = 'Matrix Partners'

In [502]:
temp2.loc[temp2['investor_name'].str.contains('Brand'),'investor_name'] = 'Brand Capital'

In [503]:
temp2.loc[temp2['investor_name'].str.contains('Angel Investors'),'investor_name'] = 'Angel Investors'

In [504]:
temp2.loc[temp2['investor_name'].str.contains('India Quotient'),'investor_name'] = 'India Quotient'

In [505]:
temp2.loc[temp2['investor_name'].str.contains('SoftBank'),'investor_name'] = 'SoftBank Group'

In [506]:
temp2.loc[temp2['investor_name'].str.contains('Info Edge'),'investor_name'] = 'Info Edge Ltd'

In [507]:
temp2.loc[temp2['investor_name'].str.contains('Bessemer Venture'),'investor_name'] = 'Bessemer Venture Partners'

In [508]:
temp2.loc[temp2['investor_name'].str.contains('Lightspeed '),'investor_name'] = 'Lightspeed Venture Partners'

In [509]:
temp2.loc[temp2['investor_name'].str.contains('Helion'),'investor_name'] = 'Helion Venture Partners'

In [510]:
temp2.loc[temp2['investor_name'].str.contains('GrowX'),'investor_name'] = 'GrowX Ventures'

In [511]:
temp2.loc[temp2['investor_name'].str.contains('Hyderabad '),'investor_name'] = 'Hyderabad Angels'

In [512]:
temp2.loc[temp2['investor_name'].str.contains('Chennai Angels'),'investor_name'] = 'Chennai Angels'

In [513]:
temp2.loc[temp2['investor_name'].str.contains('Mumbai Angels'),'investor_name'] = 'Mumbai Angels'

In [514]:
temp2.loc[temp2['investor_name'].str.contains('Nilekani'),'investor_name'] = 'Nandan Nilekani'

In [515]:
temp2.loc[temp2['investor_name'].str.contains('Mohandas'),'investor_name'] = 'TV Mohandas Pai'

In [516]:
temp2.loc[temp2['investor_name'].str.contains('Paytm'),'investor_name'] = 'Paytm'

In [517]:
temp2.loc[temp2['investor_name'].str.contains('1Crowd'),'investor_name'] = '1Crowd'

In [518]:
temp2.loc[temp2['investor_name'].str.contains('Alibaba'),'investor_name'] = 'Alibaba'

In [519]:
temp2.loc[temp2['investor_name'].str.contains('Amazon'),'investor_name'] = 'Amazon'

In [520]:
temp2.loc[temp2['investor_name'].str.contains('Ant'),'investor_name'] = 'Ant Financials'

In [521]:
temp2.loc[temp2['investor_name'].str.contains('Bee'),'investor_name'] = 'Beenext'

In [522]:
temp2.loc[temp2['investor_name'].str.contains('Bertelsmann'),'investor_name'] = 'Bertelsmann India'

In [523]:
temp2.loc[temp2['investor_name'].str.contains('Bennett'),'investor_name'] = 'Bennett'

In [524]:
temp2.loc[temp2['investor_name'].str.contains('BlackSoil'),'investor_name'] = 'BlackSoil Capital Pvt. Ltd.' 

In [525]:
temp2.loc[temp2['investor_name'].str.contains('Orchard'),'investor_name'] = 'BlueOrchard Finance Ltd'

In [526]:
temp2.loc[temp2['investor_name'].str.contains('CDC'),'investor_name'] = 'CDC Group Plc'

In [527]:
temp2.loc[temp2['investor_name'].str.contains('Cataraman'),'investor_name'] = 'Catamaran Ventures'

In [528]:
temp2.loc[temp2['investor_name'].str.contains('Cross Border'),'investor_name'] = 'Cross Border Angels'

In [529]:
temp2.loc[temp2['investor_name'].str.contains('DSG'),'investor_name'] = 'DSG Consumer Partners'

In [530]:
temp2.loc[temp2['investor_name'].str.contains('Eight'),'investor_name'] = 'Eight Roads Ventures' 

In [531]:
temp2.loc[temp2['investor_name'].str.contains('Flipkart'),'investor_name'] = 'Flipkart'

In [532]:
temp2.loc[temp2['investor_name'].str.contains('GIC'),'investor_name'] = 'GIC ventures'

In [533]:
temp2.loc[temp2['investor_name'].str.contains('Goldman'),'investor_name'] = 'Goldman Sachs'

In [534]:
temp2.loc[temp2['investor_name'].str.contains('oogle'),'investor_name'] = 'Google'

In [535]:
temp2.loc[temp2['investor_name'].str.contains('Gray Matter'),'investor_name'] = 'Gray Matters Capital'

In [536]:
temp2.loc[temp2['investor_name'].str.contains('Green House'),'investor_name'] = 'Green House Ventures'

In [537]:
temp2.loc[temp2['investor_name'].str.contains('HDFC'),'investor_name'] = 'HDFC'

In [538]:
temp2.loc[temp2['investor_name'].str.contains('HT'),'investor_name'] = 'HT Media'

In [539]:
temp2.loc[temp2['investor_name'].str.contains('IIFL'),'investor_name'] = 'IIFL Ventures'

In [540]:
temp2.loc[temp2['investor_name'].str.contains('IL&FS'),'investor_name'] = 'IL&FS'

In [541]:
temp2.loc[temp2['investor_name'].str.contains('pring'),'investor_name'] = 'IdeaSpring Capital'

In [542]:
temp2.loc[temp2['investor_name'].str.contains('InCred'),'investor_name'] = 'InCred'

In [543]:
temp2.loc[temp2['investor_name'].str.contains('Insight'),'investor_name'] = 'Insight Partners'

In [544]:
temp2.loc[temp2['investor_name'].str.contains('International Finance'),'investor_name'] = 'International Finance Corporation'

In [545]:
temp2.loc[temp2['investor_name'].str.contains('Lightbox'),'investor_name'] = 'Lightbox Ventures'

In [546]:
temp2.loc[temp2['investor_name'].str.contains('Lionrock',case=False),'investor_name'] = 'LionRock Capital'

In [547]:
temp2.loc[temp2['investor_name'].str.contains('M&S'),'investor_name'] = 'M&S Partners'

In [548]:
temp2.loc[temp2['investor_name'].str.contains('Mitsui'),'investor_name'] = 'Mitsui & Co.'

In [549]:
temp2.loc[temp2['investor_name'].str.contains(r'Business School Seed Fund.'),'investor_name'] = 'S&d Business School Seed Fund'

In [550]:
temp2['investor_name']=temp2['investor_name'].str.encode('utf-8').str.decode('unicode_escape').\
str.replace(r'\\xc2\\xa0', '', regex=True)


In [551]:
temp2.loc[temp2['startup_name'].str.contains('vidya'),'startup_name'] = 'credit vidya'

In [552]:
temp2.loc[temp2['startup_name'].str.contains('little black book'),'startup_name'] ='little black book'

In [553]:
temp2.loc[temp2['startup_name'].str.contains('ola cabs'),'startup_name'] = 'olacabs' 

In [554]:
temp2.loc[temp2['startup_name'].str.contains('oyoroom'),'startup_name'] = 'oyo rooms'

In [555]:
temp2.loc[temp2['startup_name']=='oyo','startup_name'] = 'oyo rooms'

In [556]:
temp2.loc[temp2['startup_name'].str.contains('paytm'),'startup_name'] = 'paytm'

In [587]:
temp2.loc[temp2['investor_name'].str.contains('3one4',case=False),'investor_name'] = '3one4 Capital'

In [593]:
temp2.loc[temp2['investor_name'].str.contains('50k',case=False),'investor_name'] = '50K Ventures' 


In [594]:
temp2['investor_name'].sort_values(ascending=True).unique()

array([' Sandeep Aggarwal', '021 Capita', '1Crowd', ...,
       'responsAbility Investments', 'ru-Net', 'vCommission'],
      dtype=object)

In [595]:
temp2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3044 entries, 0 to 3043
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   sno            3044 non-null   int64         
 1   date           3044 non-null   datetime64[ns]
 2   startup_name   3044 non-null   object        
 3   vertical       3044 non-null   object        
 4   subvertical    3044 non-null   object        
 5   city           3044 non-null   object        
 6   investor_name  3044 non-null   object        
 7   type           3040 non-null   object        
 8   amount         3044 non-null   float64       
dtypes: datetime64[ns](1), float64(1), int64(1), object(6)
memory usage: 214.2+ KB


In [596]:
temp2['date'] = temp2['date'].str.replace('.','-').str.replace(',','-').str.replace('//','-').str.replace('/','-').str.replace(r'\\xc2\\xa0','')

AttributeError: Can only use .str accessor with string values!

In [597]:
temp2.loc[192,'date'] = '05-07-2018'

In [598]:
temp2['date']

0      2020-01-09
1      2020-01-13
2      2020-01-09
3      2020-01-02
4      2020-01-02
5      2020-01-13
6      2020-01-10
7      2019-12-12
8      2019-12-06
9      2019-12-03
10     2019-12-13
11     2019-12-17
12     2019-12-16
13     2019-12-16
14     2019-12-14
15     2019-12-11
16     2019-12-20
17     2019-11-13
18     2019-11-14
19     2019-11-13
20     2019-11-17
21     2019-11-18
22     2019-11-15
23     2019-11-20
24     2019-11-12
25     2019-11-20
26     2019-11-11
27     2019-11-19
28     2019-11-18
29     2019-11-15
30     2019-11-19
31     2019-11-25
32     2019-10-04
33     2019-10-02
34     2019-10-21
35     2019-09-05
36     2019-09-04
37     2019-09-04
38     2019-09-04
39     2019-09-04
40     2019-09-04
41     2019-09-04
42     2019-09-04
43     2019-09-03
44     2019-08-01
45     2019-08-01
46     2019-08-01
47     2019-08-01
48     2019-08-01
49     2019-08-12
50     2019-08-13
51     2019-08-13
52     2019-08-23
53     2019-08-23
54     2019-08-23
55     201

In [599]:
temp2.loc[temp2['date']=='01-07-015','date'] = '01-07-2015' 

In [600]:
temp2['date'] = pd.to_datetime(temp2['date'],format='%d-%m-%Y')

In [601]:
temp2.to_csv('startup_cleaned.csv')